### PR TITLE
[KNIFE-331] add the ability to pass in --user-data

### PR DESCRIPTION
**Pull request by svanzoest**
http://tickets.opscode.com/browse/KNIFE-331

Hi this adds similar functionality to http://tickets.opscode.com/browse/KNIFE_EC2-3 but for Eucalyptus.

It would be great to have all the fog options available via knife.rb and CLI.

I had to remove the short -u, as that conflicts with an option that is already there.

### CODE REVIEW

**Comment by svanzoest** [comment]
Does anyone look at these at all?

**Comment by knuckolls** [comment]
Opscode is pretty strict about the contributor process noted here. Thanks for this patch. I hope you go through the steps to get it merged into master.

http://wiki.opscode.com/display/chef/How+to+Contribute

**Comment by svanzoest** [comment]
@knuckolls I am approved single contributor #72.

**Comment by sethvargo** [comment]
@svanzoest this is now http://tickets.opscode.com/browse/KNIFE-331 in JIRA and is awaiting your feedback on a few comments. Thanks :smile:!

File: lib/chef/knife/euca_server_create.rb [hunk 1]
```diff
@@ -124,6 +124,12 @@ class EucaServerCreate < Knife
         :boolean => true,
         :default => false
 
+      option :euca_user_data,
+         :long => "--user-data USER_DATA_FILE",
+         :description => "The Eucalyptus User Data file to provision the instance with",
+         :proc => Proc.new { |m| Chef::Config[:knife][:euca_user_data] = m },
```
Comment (adamedx) [inline lib/chef/knife/euca_server_create.rb:130]: So I think we should use locate_config_value here to avoid issues with precedence, etc. Anyone else?

Comment (sethvargo) [inline lib/chef/knife/euca_server_create.rb:130]: I agree

Comment (svanzoest) [inline lib/chef/knife/euca_server_create.rb:130]: fine by me, does someone want to take that on, or how would I approach that?

**Comment by knuckolls** [comment]
Been doing a real big cleanup of github. I see this never got merged. Just a bump. I vote for a merge, but closing the PR would work too. I know this code works, or at least it did two years ago when I merged it myself and got things working locally.

**Comment by knuckolls** [comment]
@btm @lamont-granquist since Seth has moved on from Chef I'm pinging you guys since you have the last two commits on this repo.

**Comment by svanzoest** [comment]
Let me know how I can help.

**Comment by knuckolls** [comment]
@adamedx pinging you as well, since you've commented on the code

**Comment by lamont-granquist** [comment]
oh lol, `--no-host-key-verify`, yeah i don't get into this repo much, its more of an @adamedx thing.

**Comment by knuckolls** [comment]
I'd love to have this just merged or closed to get it out of my github "open, mentioned, pull requests" view. 

For sure it is merely a minor annoyance / first world problem so I'll be 100% alright if it stays this way. No big. Figured i'd bump the issue just to see what might happen. 

  ¯_(ツ)_/¯

**Comment by bcarpenter** [comment]
Maybe just merging is a good plan, especially since it works?  Seems like a handy feature for EC2---I mean Eucalyptus. :+1:

**Comment by knuckolls** [comment]
bump. let's merge. :)

**Comment by knuckolls** [comment]
@nathenharvey do you know who has merge capabilities for this repo? I know it's old, but it's worth merging.

**Pull request closed without merging**